src/answer/answer.service.ts

```diff
@@ -498,4 +631,23 @@ export class AnswerService {
       userId,
     );
   }
+
+  async getAnswerCount(userId: number | undefined): Promise<number> {
+    if (userId == undefined) return 0;
+    return await this.answerRepository.countBy({ createdById: userId });
+  }
+
+  async getAnswerIdOfCreatedBy(
+    questionId: number,
+    createdById: number,
+  ): Promise<number | undefined> {
+    const answer = await this.answerRepository.findOne({
+      where: {
+        deletedAt: undefined,
+        questionId,
+        createdById,
+      },
+    });
+    return answer?.id ?? undefined;
```

Comment:
`answer?.id` is enough

Reply:
this is more clear, and nothing wrong

Reply:
fixed: https://github.com/SageSeekerSociety/cheese-backend/commit/e5b65861553c48d595edc822568a3d45e799ddff